nityam_kumar/PostgresAssignment/src/controller/mark.ts

```diff
@@ -0,0 +1,165 @@
+import { Context } from "vm";
+import * as marksService from "../services/mark";
+
+import markSchema from "../db/validateSchema/markSchema";
+import AppError from "../utils/appError";
+
+interface IMark {
+  student_id: string;
+  subject_id: string;
+  marks: number;
+  teacher_id: string;
+  class_number: number;
+}
+
+export const createMarks = async (ctx: Context) => {
+  const m1: IMark = ctx.request.body;
+  try {
+    await markSchema.validateAsync(m1);
+
+    await marksService.checkStudentExist(m1);
```

Comment:
Can we run this multiple sequence of queries in parallel ? and use Promises to resolve in parallel....

Reply:
combined checkStudentExist, checkSubjectExist, checkAlreadyExist in promise.all